cli/ast.rs

```diff
@@ -89,57 +82,18 @@ impl std::fmt::Display for Location {
   }
 }
 
-/// A buffer for collecting diagnostic messages from the AST parser.
-#[derive(Debug)]
-pub struct DiagnosticBuffer(Vec<String>);
```

Comment:
The code only possibly returns one diagnostic. Removed and replaced with a single `Diagnostic` error struct. This will also allow me to format a diagnostic specifically for the REPL (no file name).